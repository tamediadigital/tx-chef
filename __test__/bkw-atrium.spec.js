const fn = require('../helpers/bkw-atrium');

const DEBUG = true;

describe('get atrium menu', () => {
	test('exports a promise with menu data', () => {
		expect.assertions(2);
		return fn('bkw-atrium.com', DEBUG).then(data => {
			expect(data.date).toBe('21.09.');
			expect(data).toMatchSnapshot();
		});
	});

	test('finds the first meal', () => {
		expect.assertions(5);
		return fn('bkw-atrium', DEBUG).then(data => {
			const meal = Object.values(data.meals)
				.filter(mealObj => mealObj.title === 'Grilliertes MSC Rotbarschfilet')
				.pop();

			expect(meal.description).toBe('mit Zitronensauce, Couscous und Mangold mit Rahmsauce Add on: 1 ASC Krevettenspiess CHF 3.50');
			expect(meal.provenance).toBe('Herkunft: Krevetten: Vietnam, Zucht, Rotbarsch, Wildfang, Island');
			expect(meal.prices).toMatchObject(['13.50 CHF', '9.50 CHF']);
			expect(meal.vegetarian).toBe(false);
			expect(meal.vegan).toBe(false);
		});
	});

	test('finds vegetarian meals', () => {
		expect.assertions(2);
		return fn('bkw-atrium', DEBUG).then(data => {
			const meal = Object.values(data.meals)
				.filter(mealObj => mealObj.title === 'Fusilli Nikos')
				.pop();

			expect(meal.vegetarian).toBe(true);
			expect(meal.vegan).toBe(false);
		});
	});
});
