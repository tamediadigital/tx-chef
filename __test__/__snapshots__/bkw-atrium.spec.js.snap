// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get atrium menu exports a promise with menu data 1`] = `
Object {
  "date": "21.09.",
  "meals": Object {
    "0": Object {
      "category": "",
      "description": "mit Zitronensauce, Couscous und Mangold mit Rahmsauce Add on: 1 ASC Krevettenspiess CHF 3.50",
      "id": "0",
      "prices": Array [
        "13.50 CHF",
        "9.50 CHF",
      ],
      "provenance": "Herkunft: Krevetten: Vietnam, Zucht, Rotbarsch, Wildfang, Island",
      "title": "Grilliertes MSC Rotbarschfilet",
      "vegan": false,
      "vegetarian": false,
    },
    "1": Object {
      "category": "",
      "description": "Vollkornpasta mit Pesto, Feta, Oliven, Cherrytomaten und Spinatblätter Add on: Samosas Stk. CHF 2.00",
      "id": "1",
      "prices": Array [
        "13.50 CHF",
        "9.50 CHF",
      ],
      "provenance": "",
      "title": "Fusilli Nikos",
      "vegan": false,
      "vegetarian": true,
    },
    "2": Object {
      "category": "",
      "description": "mit Hackfleisch im Fladenbrot mit Joghurt-, oder Cocktailsauce inklusive 5 dl SV Ice Tea ... es het, solang's het ...",
      "id": "2",
      "prices": Array [
        "16.90 CHF",
        "12.90 CHF",
      ],
      "provenance": "Poulet, Schweiz, Rind, Schweiz, Kalb, Schweiz",
      "title": "Dürüm",
      "vegan": false,
      "vegetarian": false,
    },
    "3": Object {
      "category": "",
      "description": "mit Waldpilzsauce, Semmelknödel und Rotkraut mit Sultaninen",
      "id": "3",
      "prices": Array [
        "13.50 CHF",
        "9.50 CHF",
      ],
      "provenance": "Schwein, Schweiz",
      "title": "Schweins- Schulter- Braten",
      "vegan": false,
      "vegetarian": false,
    },
  },
}
`;
